### DISCUSSION TITLE
How to use with Gin?

### DISCUSSION THREAD

**Discussion by gaetandezeiraud** [go-pkgz/auth#196] [Q&A] [answered]
Hello!
I want to implement this package in my Go project. But I use Gin. I fund this open source code https://github.com/katomaso/gin-auth/blob/main/lib.go who do some abstraction. But I have a problem, when I do `Router.GET("/auth", auth.AuthHandler())` with AuthHandler as follow :

```go
func (s *Service) AuthHandler() gin.HandlerFunc {
	authHandler, _ := s.Handlers()
	return func(c *gin.Context) {
		authHandler.ServeHTTP(c.Writer, c.Request)
	}
}
```

But routes like /auth/list or providers doesn't exist. I have a 404 page not found.
Any idea?

Thanks!

**Comment by alek-sys** [comment, marked as answer]
You don't need to define your own handler, Gin has a wrapper to use standard `net/http` compatible handlers (which is what `auth` provides):
```go
authHandler, _ := s.Handlers()
r := gin.Default()
r.Group("/auth").GET("/*path", gin.WrapH(authHandler))
```

**Reply by gaetandezeiraud** [reply]
Thank you a lot!
